### PR TITLE
core/types: SigCache with a limited size

### PR DESCRIPTION
**Pull request by jimmyhu-dexon**

### CODE REVIEW

**Comment by aitjcize** [comment]
Nice, this is exactly what I'm talking about. Have you run trace against this?

File: core/types/transaction_signing.go [hunk 1]
```diff
@@ -94,8 +101,22 @@ func (c *globalSigCache) Add(signer Signer, txs Transactions) (errorTx *Transact
...
+				var prune []resultEntry
+				c.cache, prune = c.cache[:pruneTargetSize], c.cache[pruneTargetSize:]
+				for _, r := range prune {
+					delete(c.cacheID, r.Hash)
+				}
+			}
 			for _, r := range results {
-				c.cache[r.Hash] = r.Addr
+				c.cacheID[r.Hash] = len(c.cache)
+				c.cache = append(c.cache, r)
```
Comment (aitjcize) [inline core/types/transaction_signing.go]: one useful speedup is to makesure c.cache is big enough for all the results. This makes a huge difference.

Comment (jimmyhu-dexon) [reply core/types/transaction_signing.go]: `c.cache` is allocated with size `sigCacheSize`

Comment (aitjcize) [reply core/types/transaction_signing.go]: oh got it.

**Pull request merged**